### PR TITLE
Parsing

### PR DESCRIPTION
**Pull request by CalebCourier**
This PR addresses the original issue captured in https://github.com/ShreyaR/guardrails/issues/270 as well as the underlying issue in https://github.com/ShreyaR/guardrails/issues/226.
It does not address the similar issue describe in the comments.

The implementation for this attempts to reuse the existing flows for reasks and successfully does so in the integration tests. We do however need to test if the new parsing specific reask prompt works well for this scenario for getting the LLM to fix its mistakes.

### CODE REVIEW

**Comment by CalebCourier** [comment]
@zsimjee Odd behavior on the Github Actions here.  The tests are running and passing but the check is failing because it's failing to upload to codecov.io.

> [2023-09-06T14:05:28.741Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255

Edit: Looks like a token issue.  Is there a secret set for CODECOV_TOKEN? If so is it expired?

**Comment by irgolic** [comment]
> [2023-09-06T14:05:28.741Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
Error: Codecov: Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v3/dist/codecov' failed with exit code 255

I get this a lot, should work if you wait a bit and rerun it

File: guardrails/utils/parsing_utils.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        string_value (str): The string to check.
+        code_type (str, optional): The specific code type to check for. Defaults to empty string.
+
+    Returns
+        bool: Whether or not the string contains the specified type of code block.
+        int: The starting index of the code block.
+        int: The ending index of the code block.
+    """  # noqa
+    block_border = "```"
+    block_start_border = f"{block_border}{code_type}"
```
Comment (zsimjee) [inline guardrails/utils/parsing_utils.py:20]: is code type always present?

**Review by zsimjee — DISMISSED** [review]

File: guardrails/utils/parsing_utils.py (hunk 1 above)
Comment (CalebCourier) [reply guardrails/utils/parsing_utils.py:20]: No, but it defaults to empty string

Comment (CalebCourier) [reply guardrails/utils/parsing_utils.py:20]: So effectively if code_type is passed `block_start_border` is "```json" or "```js" etc. if not then it's just "```"

**Review by zsimjee — APPROVED** [review]

**Pull request merged**